webpush/static/webpush/webpush_serviceworker.js

```diff
@@ -1,20 +1,34 @@
-
+let url;
 // Register event listener for the 'push' event.
 self.addEventListener('push', function(event) {
   // Retrieve the textual payload from event.data (a PushMessageData object).
   // Other formats are supported (ArrayBuffer, Blob, JSON), check out the documentation
   // on https://developer.mozilla.org/en-US/docs/Web/API/PushMessageData.
-  var payload = event.data ? event.data.text() : {"head": "No Content", "Body": "No Content"},
+  let payload = event.data ? event.data.text() : {"head": "No Content", "body": "No Content", "icon": "", "url": ""},
     data = JSON.parse(payload),
     head = data.head,
-    body = data.body;
+    body = data.body,
+    icon = data.icon;
+  // Url needs to be acessed from outside this method.
+  url = data.url
```

Comment:
Do you mean, do this?
``` javascript
    body = data.body,
    icon = data.icon,
    url = data.url;
```
url would be undefined in the notificationonclick if I do that. Maybe because I'm creating a new variable or something? I don't know. Anyway, in the pull, the indent is in the same level than the payload, with makes more sense.